Работа с датами и приоритетами.

masmetus opened the pull request.

masmetus committed:
Работа с датами и приоритетами.

masmetus committed:
fix after CheckStyle

z8q commented on src/manager/InMemoryTaskManager.java:
```diff
@@ -5,27 +5,33 @@
+    final Map<Integer, Subtask> subtaskMap = new HashMap<>();
+    final Map<Integer, Epic> epicMap = new HashMap<>();
+    final Map<Integer, Task> taskMap = new HashMap<>();
+    final HistoryManager historyManager = new InMemoryHistoryManager();
```
можно поставить protected на поля, чтобы ими могли пользоваться только классы наследники

иначе получится, что мы можем работать с публичными мапами в обход менеджера,
не через методы, в которых есть проверки и правила добавления тасок, а напрямую

z8q commented on src/manager/FileBackedTaskManager.java:
```diff
@@ -19,49 +21,71 @@ public FileBackedTaskManager(File file) {
+                        manager.epicMap.put(epic.getId(), epic);
+                    } else if (task instanceof Subtask subtask) {
+                        manager.subtaskMap.put(subtask.getId(), subtask);
+                        Epic epic = manager.epicMap.get(subtask.getEpicId());
+                        if (epic != null) {
+                            epic.addSubtaskId(subtask.getId());
+                        }
```
тут в 70-й строке добавляем саб-таску,
и потом проверяем, есть ли эпик, если эпика нет, то не связываем саб-таску с эпиком

по идее, в менеджере не должно быть саб-тасок, которые не привязаны к эпику

но допускаю, что у нас в файле может быть рандомный порядок тасок, и, когда добавляем саб-таску, может быть, что эпик, к которому ее надо привязать, еще не добавлен

возможно сначала стоит добавить все таски/саб-таски/эпики, чтобы мы точно знали, что все добавлено,
и потом пройтись по значениям из мапы саб-тасок и привязать их к эпикам,
если эпиков для привязки нет, то может удалить такие саб-таски, чтобы сохранить консистентное состояние в менеджере
можно продумать этот момент

masmetus committed:
fix after review:
1. Изменена логика работы с загрузкой задачи
2. Модификатор доступа у мап с данными
3. Тест на новую логику

masmetus committed:
minor fix
